### DISCUSSION TITLE
Search within relation content type

### DISCUSSION THREAD

**Discussion by asdutoit** [strapi/strapi#8562] [Legacy Topics]
Hi There, 

I have a simple content-type "Onts".   "Onts" has a few relationships to other content-type, for example, "address_unit" or "services".   See screenshot below.

![Screenshot 2020-11-03 at 16 18 54](https://user-images.githubusercontent.com/20240370/97996378-56c46480-1df0-11eb-9333-a3b34f0a5146.png)
![Screenshot 2020-11-03 at 16 19 06](https://user-images.githubusercontent.com/20240370/97996395-5d52dc00-1df0-11eb-96c8-4992a28799cd.png)

When I run a search query with Postman or the built-in Swagger platform, I successfully retrieve search results from any fields within the "Onts" document, however, I am unable to search within the relational documents.
See screenshots below.   
First Screenshot shows that I have success in searching for "5A594F43806616F8".   
However, with 2nd screenshot, I am trying to search "106" (which is the "unit_no" in "address_unit") but with no success. 
I have also tried searching for "5f01f63a408c5c5d59a04037" (which is the "vlan" in "services") with no success.  

![Screenshot 2020-11-03 at 16 21 31](https://user-images.githubusercontent.com/20240370/97996855-f255d500-1df0-11eb-9240-285c82a852f4.png)
![screenshot2](https://user-images.githubusercontent.com/20240370/97996956-16b1b180-1df1-11eb-8442-144429767637.png)

I understand I will need to build a custom controller and services function in Strapi.   

This is my current "/api/onts/services/onts.js" file:
```javascript
module.exports = {
  search: async (params) => {
    return strapi.query("onts").search(params);
  },
};
```

and this is my /api/onts/controllers/onts.js" file:
```javascript
module.exports = {
  async find(ctx) {
    let entities;
    if (ctx.query._q) {
      entities = await strapi.services.onts.search(ctx.query);
    } else {
      entities = await strapi.services.onts.find(ctx.query);
    }

    return entities.map((entity) =>
      sanitizeEntity(entity, { model: strapi.models.onts })
    );
  },
};
```

How do I go about in modifying the code to be able to search in relational documents as well?

Any assistance is appreciated. 

Warm Regards,

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Reply by asdutoit** [reply]
Hi @derrickmehaffy , Sure I will ask on the forum as well
